controllers/class.js

```diff
@@ -146,57 +162,105 @@ exports.write_to_csv = (req, res, next) => {
 				element.picture,
 			]);
 		});
-		
+
 		// TODO - make filenames more descriptive
 		csv_writer
 			.writeToPath("uploads/csv/students.csv", values, {headers: true})
 			.on("finish", send_response);
 	}
-	
+
 	function send_response(err, result, args){
 	 	if(err){
 	 		winston.error('Could not write to CSV');
 	 		return next(err);
 	 	}
-	 	
+
 	 	res.send();
 	 }
-	
+
 	start();
 };
 
 exports.insert_csv_classlist = (req, res, next) => {
 
-    function start () {
-		
-		let class_query;
-		
-		sh.cd('controllers');
-		sh.config.silent = true;
-		sh.exec('sudo chmod 755 ../helpers/classlist.js');
-		sh.config.silent = false;
-		sh.exec('node ../helpers/classlist.js ../uploads/csv/classlist.csv > ../database/classlist.sql', function (err) {
-
-			if (err) {
-				winston.error('Error in inserting classlist from CSV');
-				return next(err);
-			}
+	let path;
 
-		});
-		class_query = sh.exec('cat ../database/classlist.sql').output;
-		
-		// TODO - convert to formal query
-		res.send();
+	function start () {
+		upload(req, res, initialize_shell);
+	}
 
-    }
-	
-	function send_response (err, result, args, last_query) {
+	function initialize_shell (err) {
+		if(err) {
+			winston.error('Error uploading file.');
+			return next(err);
+		}
+
+		path = req.file.path;
+		sh.exec('cd ./', generate_container);
+	}
+
+	function generate_container (err) {
+		if (err) {
+			winston.error('Error in initializing shell');
+			return next(err);
+		}
+
+		sh.exec('touch uploads/csv', overwrite_permission);
+	}
+
+	function overwrite_permission (err) {
+		if (err) {
+			winston.error('Error in generating csv folder');
+			return next(err);
+		}
+
+		sh.exec('sudo chmod 755 helpers/classlist.js', generate_query);
```

Comment:
Just put this step into setup or at the start of server.js . Nauulit siya kahit di naman kailangan